library/node

```diff
@@ -4,6 +4,41 @@ Maintainers: The Node.js Docker Team <https://github.com/nodejs/docker-node> (@n
 GitRepo: https://github.com/nodejs/docker-node.git
 GitFetch: refs/heads/main
 
+Tags: 16.0.0-alpine3.11, 16.0-alpine3.11, 16.0.0-alpine, 16.0-alpine
+Architectures: amd64, arm32v6, arm32v7, arm64v8, ppc64le, s390x
+GitCommit: f0551c9eb5e0dbb853eda4266bc0964e5989bda1
+Directory: 16/alpine3.11
+
+Tags: 16.0.0-alpine3.12, 16.0-alpine3.12
+Architectures: amd64, arm32v6, arm32v7, arm64v8, ppc64le, s390x
+GitCommit: f0551c9eb5e0dbb853eda4266bc0964e5989bda1
+Directory: 16/alpine3.12
+
+Tags: 16.0.0-alpine3.13, 16.0-alpine3.13
+Architectures: amd64, arm32v6, arm32v7, arm64v8, ppc64le, s390x
+GitCommit: f0551c9eb5e0dbb853eda4266bc0964e5989bda1
+Directory: 16/alpine3.13
+
+Tags: 16.0.0-buster, 16.0-buster
+Architectures: amd64, arm32v7, arm64v8, ppc64le, s390x
+GitCommit: 2438859c181e4e9e5c4a7ad8e4a54ef69aeefbf3
+Directory: 16/buster
+
+Tags: 16.0.0-buster-slim, 16.0-buster-slim
+Architectures: amd64, arm32v7, arm64v8, ppc64le, s390x
+GitCommit: 2438859c181e4e9e5c4a7ad8e4a54ef69aeefbf3
+Directory: 16/buster-slim
+
+Tags: 16.0.0-stretch, 16.0-stretch, 16.0.0, 16.0
```

Comment:
Still keeping `stretch` rather than `buster` as the default for the new release (`16.0` and `16.0-slim`)? And same comment for `16.0-alpine` being Alpine `3.11` instead of `3.13`.

Also, did you want to move the generic tags like `latest`, `current`, `buster`, `alpine`, and related tags from `15` images to `16` images?  Also missing tags like `16`, `16-slim`, `16-alpine`.

Reply:
> Still keeping `stretch` rather than `buster` as the default for the new release (`16.0` and `16.0-slim`)? And same comment for `16.0-alpine` being Alpine `3.11` instead of `3.13`.

Still not untangled in the repo, as the default versions are still global currently rather than version specific



> Also, did you want to move the generic tags like `latest`, `current`, `buster`, `alpine`, and related tags from `15` images to `16` images? Also missing tags like `16`, `16-slim`, `16-alpine`.

Right, I'll submit a PR to update the `generate` script to include 16. I think that's what was missing for that part

Reply:
Actually, I think we still have a per-directory config, we just haven't been using it. I'll try and add it to the PR and see

Reply:
@yosifkit I fixed the current/latest bit. For the proper per-version defaults I think the generate-stackbrew setup needs to be re-aligned. Is there a particular one to look at as a basis? I tried looking at the golang one, but I didn't quite get it running

Reply:
When supporting more than one OS suite (or upstream language version), we usually just make an array of the defaults for that version of the software along with a global fallback variable. Here is an example of drupal that has default Debian suite, Alpine version, and PHP versions:

(https://github.com/docker-library/drupal/blob/680831527ba3ff1aa7376331a1a427060ef852de/generate-stackbrew-library.sh#L10-L22)
```bash
defaultDebianSuite='buster'
declare -A debianSuites=(
	#[9.0]='buster'
)
defaultAlpineVersion='3.12'

defaultPhpVersion='php8.0'
declare -A defaultPhpVersions=(
# https://www.drupal.org/docs/7/system-requirements/php-requirements#php_required
	[7]='php7.4'
	[8.9]='php7.4'
	[9.0]='php7.4'
)
```

And then we use it like this to get the default suite for the version:
```bash
debianSuite="${debianSuites[$version]:-$defaultDebianSuite}"
```

----------------

You probably just need something like the following in the start of the loop on [line 62](https://github.com/nodejs/docker-node/blob/13106300db1b4ce862010e3b35b86cc25cbe4cf5/generate-stackbrew-library.sh#L62), swap uses of `default_variant` for `version_default_variant` in the loop and then add a `config` file in each version folder that wants to change the default.

```bash
version_default_variant="$(get_config "./$version" "default_variant")"
```